src/ParquetMsg.chpl

```diff
@@ -1304,22 +1304,41 @@ module ParquetMsg {
           const high = saD.high;
           const locDom = sa.localSubdomain();
           var values = getGenericTypedArrayEntry(components["values"], st);
-          
+
+          segment_ct[i] += locDom.size;
+          if values.dtype == DType.Strings && locDom.size > 0 {
+            var e: SegStringSymEntry = toSegStringSymEntry(values);
+            var segStr = new SegString("", e);
+            ref ss = segStr;
+            var lens = ss.getLengths();
+            const lastOffset = if sa.size == 0 then 0 else sa[high];
+            const lastOffsetIdx = segStr.offsets.a.domain.high;
+            var startOffsetIdx = sa[locDom.low];
+            var endOffsetIdx = if (lastOffset == sa[locDom.high]) then lastOffsetIdx else sa[locDom.high + 1] - 1;
+            var offIdxRange = startOffsetIdx..endOffsetIdx;
+            var str_bytes: int;
+            for i in offIdxRange do str_bytes += lens[i];
```

Comment:
it probably doesn't matter (and we do it in other places) but for clarity maybe we should change the looping variable of this `for` loop to be different from the outer `forall` loop. They both use `i` and it threw me off for a sec. It's up to you though, I don't think this should affect the functionality at all